src/main/java/io/fabric8/maven/docker/wait/ExitCodeChecker.java

```diff
@@ -0,0 +1,35 @@
+package io.fabric8.maven.docker.wait;
+
+import io.fabric8.maven.docker.access.DockerAccessException;
+import io.fabric8.maven.docker.service.ServiceHub;
+
+public class ExitCodeChecker implements WaitChecker {
+
+    private final int exitCode;
+    private final ServiceHub hub;
+    private final String containerId;
+
+    public ExitCodeChecker(int exitCode, ServiceHub hub, String containerId) {
```

Comment:
Could you please add a unit test for the checker ? You can use jmockit for mocking the service hub, please have a look at the existing tests.